compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirCallCompletionResultsWriterTransformer.kt

```diff
@@ -223,49 +221,29 @@ class FirCallCompletionResultsWriterTransformer(
         return transformElement(block, data)
     }
 
-    override fun transformWhenExpression(whenExpression: FirWhenExpression, data: Nothing?): CompositeTransformResult<FirStatement> {
-        val calleeReference = whenExpression.calleeReference as? FirNamedReferenceWithCandidate ?: return whenExpression.compose()
+    // Transformations for synthetic calls generated by FirSyntheticCallGenerator
 
-        val whenExpression = whenExpression.transformChildren(this, data) as FirWhenExpression
+    override fun transformWhenExpression(whenExpression: FirWhenExpression, data: Nothing?) = transformSyntheticCall(whenExpression)
+    override fun transformTryExpression(tryExpression: FirTryExpression, data: Nothing?) = transformSyntheticCall(tryExpression)
+    override fun transformCheckNotNullCall(checkNotNullCall: FirCheckNotNullCall, data: Nothing?) = transformSyntheticCall(checkNotNullCall)
 
-        val declaration = whenExpression.candidate()?.symbol?.fir as? FirMemberFunction<*> ?: return whenExpression.compose()
+    private inline fun <reified D> transformSyntheticCall(syntheticCall: D): CompositeTransformResult<FirStatement>
```

Comment:
Refactored duplicate code here. Can we also de-dupe code in `transformFunctionCall` as well?

Reply:
We want refactor that place too

Reply:
> Can we also de-dupe code in transformFunctionCall as well?

We also want refactor this place, but don't decide what is the better way to do

Reply:
> We also want refactor this place, but don't decide what is the better way to do

Sorry, I don't understand... Can you clarify what you mean by "don't decide"? Would you like `transformFunctionCall` refactored as well, to remove the duplicate code?

Reply:
Sorry for my english
I meant that we want unify code in code resolver, but didn't decide yet, how to change tree nodes for that
There is an open question about annotation calls that, probably, should be resolved via FirCallResolver, so we don't refactor resolver and tree before we design annotations

Reply:
Okay, I will leave this alone then. It makes sense to unify in the call resolver.